### PR TITLE
refactor(lanelet2_map_validator): move headers to include/

### PR DESCRIPTION
**Pull request by soblin**
## Description

move header files to `include`/

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by TaikiYamada4 — CHANGES_REQUESTED** [review]
The refactoring regarding the include files seems good, but let me point out two things about the `validation.cpp` related stuff.

File: map/autoware_lanelet2_map_validator/src/common/validation.cpp [hunk 1]
```diff
@@ -27,31 +27,30 @@ namespace autoware
...
     return std::make_unique<lanelet::projection::TransverseMercatorProjector>(
-      lanelet::Origin{val_config.origin});
-  } else if (config.projector_type == projector_names::utm) {
-    return std::make_unique<lanelet::projection::UtmProjector>(lanelet::Origin{val_config.origin});
+      lanelet::Origin{origin});
+  } else if (projector_type == projector_names::utm) {
+    return std::make_unique<lanelet::projection::UtmProjector>(lanelet::Origin{origin});
   }
-  return std::make_unique<lanelet::projection::MGRSProjector>();
+  return nullptr;
```
[the remark this answers is not in the document]
Comment (soblin) [reply map/autoware_lanelet2_map_validator/src/common/validation.cpp:41]: Okay, but it is valid/normal that the user does not provide the projector information?

[the remark this answers is not in the document]
Comment (TaikiYamada4) [reply map/autoware_lanelet2_map_validator/src/common/validation.cpp:41]: Hmm..., I guess most users use MGRS so I thought setting it as default is not bad.
However, letting the users notice what projection they're using also feels nice.

I'm OK with making the `--projection` argument mandatory, but if that's the case, please change it like this or it just returns a segmentation fault.
```suggestion
  throw std::invalid_argument("No map projection type specified!");
```

[the remark this answers is not in the document]
Comment (soblin) [reply map/autoware_lanelet2_map_validator/src/common/validation.cpp:44]: I added nullptr check for projector/map and add error string in such cases.

File: map/autoware_lanelet2_map_validator/src/main.cpp [hunk 2]
```diff
@@ -54,7 +55,9 @@ void process_requirements(
       temp_validator_config.command_line_config.validationConfig.checksFilter = validator_name;
 
       std::vector<lanelet::validation::DetectedIssues> temp_issues =
-        lanelet::autoware::validation::validateMap(temp_validator_config);
+        lanelet::autoware::validation::validateMap(
+          validator_config.projector_type, validator_config.command_line_config.mapFile,
+          validator_config.command_line_config.validationConfig);
```
[the remark this answers is not in the document]
Comment (soblin) [reply map/autoware_lanelet2_map_validator/src/main.cpp]: I added a function `replace_validator` that return a new ValidatonConfig object with a specified validator and rest of the elements as is.

File: map/autoware_lanelet2_map_validator/src/common/validation.cpp [hunk 3]
```diff
@@ -65,7 +76,6 @@ std::vector<lanelet::validation::DetectedIssues> validateMap(const MetaConfig &
                         }));
   }
 
-  appendIssues(issues, lanelet::validation::validateMap(*map, val_config));
```
Comment (soblin) [inline map/autoware_lanelet2_map_validator/src/common/validation.cpp:68]: I moved this to try block because map was not nullptr checked.

**Review by TaikiYamada4 — COMMENTED** [review]
@soblin 
I realized that we should keep the `include` directory *inside* the `src` directory to follow the coding guidelines, unless we want to export the hpp files as libraries for others.
https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/directory-structure/#not-exporting-headers

Could you replace them please? You can also remove the `lanelet2_map_validator` layer this way.

**Review by TaikiYamada4 — APPROVED** [review]
@soblin 
Thanks for your revision!!
LGTM :+1:

**Pull request merged**